aten/src/ATen/native/sparse/cuda/SparseCsrTensorMath.cu

```diff
@@ -424,18 +425,28 @@ Tensor reduce_sparse_csr_dim0_cuda_template(const Tensor& sparse, ReductionOp ro
   std::tie(new_col_indices, std::ignore) = at::_unique(col_indices, true, false);
   auto new_nnz = new_col_indices.numel();
   Tensor new_crow_indices = at::tensor(ArrayRef<int64_t>{0, new_nnz}, col_indices.options());
-  Tensor new_values = at::empty({new_nnz}, values.options());
-
+  Tensor new_values, new_values_acc;
+
+  // Set `is_cuda` = `true` in acc_type in CPU backend. Because the accumulate type
+  // of float should be float in current scenario. In CUDA, float is the accumulate type
+  // of float, while in CPU, double is the accumulate type of float.
+  using acc_t = at::acc_type<scalar_t, true>;
+  constexpr bool need_acc = !std::is_same<scalar_t, acc_t>::value;
+  bool is_integral = at::isIntegralType(values.scalar_type(), /*includeBool=*/true);
```

Comment:
This line is repeated 4 times. The intent of my higher level comment was not around a single block of particular code. It was to abstract away repeated code.